tests/e2e/mapper/bluetooth/bluetooth_suite_test.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bluetooth
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/api/core/v1"
+
+	"github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/apis/devices/v1alpha1"
+	"github.com/kubeedge/kubeedge/tests/e2e/utils"
+)
+
+//context to load config and access across the package
+var (
+	ctx          *utils.TestContext
+	cfg          utils.Config
+	nodeSelector string
+	nodeName     string
+)
+
+var (
+	MockHandler         = "/apis/devices.kubeedge.io/v1alpha1/namespaces/default/devicemodels"
+	MockInstanceHandler = "/apis/devices.kubeedge.io/v1alpha1/namespaces/default/devices"
+	CrdHandler          = "/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions"
+	AppHandler          = "/api/v1/namespaces/default/pods"
+	NodeHandler         = "/api/v1/nodes"
+	DeploymentHandler   = "/apis/apps/v1/namespaces/default/deployments"
+	deviceCRD           = "devices.devices.kubeedge.io"
+	deviceModelCRD      = "devicemodels.devices.kubeedge.io"
+)
+
+type Token interface {
+	Wait() bool
+	WaitTimeout(time.Duration) bool
+	Error() error
+}
+
+// Testing the basic bluetooth mapper functionalities
+func TestMapperCharacteristics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	var _ = BeforeSuite(func() {
+		utils.InfoV6("Before Suite Execution")
+		cfg = utils.LoadConfig()
+		ctx = utils.NewTestContext(cfg)
+		t := time.Now()
+		nodeName = "edge-node-bluetooth"
+		nodeSelector = "node-" + utils.GetRandomString(3)
+
+		//Generate Cerificates for Edge and Cloud nodes copy to respective folders
+		Expect(utils.GenerateCerts()).Should(BeNil())
+		//Do the neccessary config changes in Cloud and Edge nodes
+		Expect(utils.DeploySetup(ctx, nodeName, "deployment")).Should(BeNil())
+
+		//Apply CRD for devicemodel
+		curPath := getpwd()
+		deviceModelPath := "../../../../build/crds/devices/devices_v1alpha1_devicemodel.yaml"
```

Comment:
Can we define the deviceModelPath and deviceInstnce path as constants ?

Reply:
Also i think it will be better if we create a model and instance in the source code ?

Reply:
applicable to similar places

Reply:
Updated as constants